metagraph/src/annotation/row_diff_builder.cpp

```diff
@@ -16,6 +16,7 @@ const uint64_t BLOCK_SIZE = 1 << 25;
 const uint64_t BUFFER_SIZE = 1024 * 1024; // 1 MiB
 const uint64_t ROW_REDUCTION_WIDTH = 32;
 const uint32_t MAX_NUM_FILES_OPEN = 2000;
```

Comment:
is there a way to read the `ulimit` from the system so this doesn't have to be hard-coded?

Reply:
the default ulimit is 1024, so this could cause problems if you eventually open 2000 files at once

Reply:
I thought the default is 4096.

Reply:
one can increase the limit to 4096 without root privileges (unless someone changes this), but the default base value is 1024

Reply:
OK, it's 4k on the cluster. I wouldn't like to change this, it has worked so far.

We can add this parameter when someone faces the issue.

Reply:
Ahh, I see where the confusion is. On our login nodes, it's 1024, but on the compute nodes, it's 4096...